src/core/func-core-tools.ts

```diff
@@ -0,0 +1,217 @@
+  if (!fs.existsSync(folder)) {
+    return undefined;
+  }
+
+  const versionFile = path.join(folder, VERSION_FILE);
+  if (!fs.existsSync(versionFile)) {
+    // Cannot detect downloaded version, in doubt cleanup the folder
+    removeDownloadedCoreTools(targetVersion);
+    return undefined;
+  }
+
+  return fs.readFileSync(versionFile, 'utf8');
+}
+
+function getPlatform() {
+  switch (os.platform()) {
+    case 'win32':
+      return 'Windows';
+    case 'darwin':
+      return 'MacOS';
+    case 'linux':
+      return 'Linux';
+    default:
+      throw new Error(`Unsupported platform: ${os.platform()}`);
+  }
+}
+
+export async function getLatestCoreToolsRelease(targetVersion: number): Promise<CoreToolsRelease> {
+  try {
+    const response = await fetch(RELEASES_FEED_URL);
+    const feed = await response.json();
+    const tag = feed.tags[`v${targetVersion}`];
+    if (!tag || tag.hidden) {
+      throw new Error(`Cannot find the latest version for v${targetVersion}`);
+    }
+
+    const release = feed.releases[tag.release];
+    if (!release) {
+      throw new Error(`Cannot find release for ${tag.release}`);
+    }
+
+    const coreTools = release.coreTools.filter((t: CoreToolsZipInfo) => t.size === 'full');
+    const platform = getPlatform();
+    const info = coreTools.find((t: CoreToolsZipInfo) => t.OS === platform);
+    if (!info) {
+      throw new Error(`Cannot find download package for ${platform}`);
+    }
+    
+    return {
+      version: tag.release,
+      url: info.downloadLink
+    };
+  } catch (error: unknown) {
+    throw new Error(`Error fetching Function Core Tools releases: ${(error as Error).message}`);
+  }
+}
+
+async function downloadAndUnzipPackage(url: string, dest: string) {
+  const response = await fetch(url);
+  const totalSize = Number(response.headers.get('content-length'));
+  const progressBar = new cliProgress.Bar({
+    format: '{bar} {percentage}% | ETA: {eta}s',
+  }, cliProgress.Presets.shades_classic);
+  let downloadedSize = 0;
+  let since = Date.now();
+  progressBar.start(totalSize, downloadedSize);
+
+  response.body.on('data', (chunk) => {
+    downloadedSize += chunk.length
+    const now = Date.now();
+    if (now - since > 100) {
+      progressBar.update(downloadedSize);
+      since = now;
+    }
+  });
+
+  await new Promise((resolve, reject) => {
+    const unzipperInstance = unzipper.Extract({ path: dest });
+    unzipperInstance.promise().then(resolve, reject);
+    response.body.pipe(unzipperInstance);
+  });
+}
+
+export async function downloadCoreTools(version: number): Promise<string> {
+  const dest = getCoreToolsFolder(version);
+  const release = await getLatestCoreToolsRelease(version);
+
+  // Make sure we start from a clean folder
+  if (fs.existsSync(dest)) {
+    removeDownloadedCoreTools(version);
+  } else {
+    fs.mkdirSync(dest, { recursive: true });
+  }
+
+  await downloadAndUnzipPackage(release.url, dest);
```

Comment:
We need to make sure that the binary is not corrupted. Can we check the SHA2 sum against the one from the feed definition?